hashes/contrib/test_vars.sh

```diff
@@ -5,10 +5,10 @@
 # shellcheck disable=SC2034
 
 # Test all these features with "std" enabled.
-FEATURES_WITH_STD="io serde small-hash schemars"
+FEATURES_WITH_STD="bitcoin-io serde small-hash schemars"
 
 # Test all these features without "std" enabled.
-FEATURES_WITHOUT_STD="alloc serde small-hash"
+FEATURES_WITHOUT_STD="alloc bitcoin-io serde small-hash"
```

Comment:
FTR I think that weak dependencies changed how features work and we'll need to refactor this, perhaps also remove it and generate the feature list automatically.

Reply:
I think it should strictly simplify things because it eliminates redundant possibilities (e.g. `hashes-std` without `hashes` is no longer possible because there is no `hashes-std`) but doesn't add any new possibilities. So the existing code should just keep working.

But I agree in general we should try to generate this automatically.

In my local CI I take the union of all explicit features and optional dependencies (though I should double-check that this logic is still correct with the new `dep:` syntax and weak dependencies). Then for crates that need a no-std feature, for every resulting feature combination I check whether `default` or `std` are in them, and if not, I jam `no-std` in there.

Reply:
Exactly. That algorithm sounds correct. I think we could get `cargo` to output the list of features correctly.